File: src/main/java/appbuilder/api/classes/While.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package appbuilder.api.classes;

import appbuilder.api.methods.Método;

/**
 *
 * @author psilva
 */
public class While {
    private String condição= "";
    private String corpo= "";
    
    public While(String condição ){
        this.condição = condição;
    }

    public String getCondição() {
        return condição;
    }

    public void setCondição(String condição) {
        this.condição = condição;
    }
    
    public void addCorpo(String corpo ){
        this.corpo += Método.formatar(corpo);
    }

    public String getCorpo() {
        return corpo;
    }

    public void setCorpo(String corpo) {
        this.corpo = Método.formatar(corpo);
    }
    
    
    
    
    public String toString(){
        String codigo = "";
        
        codigo+="while("+condição+"){\n\n";
        codigo+=corpo;
        codigo+="\t}\n";
        
        return codigo;
    }
}

File: AppBuilder/AppBuilder/Old/Classes/While.cs
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

namespace Api.Old.Classes;

/// <summary>
/// </summary>
/// <remarks>@authorpsilva</remarks>
public class While
{
    private string condição = "";
    private string corpo = "";

    public While(string condição)
    {
        this.condição = condição;
    }

    public virtual string GetCondição()
    {
        return condição;
    }

    public virtual void SetCondição(string condição)
    {
        this.condição = condição;
    }

    public virtual void AddCorpo(string corpo)
    {
        this.corpo += MethodBuilder.FormatCode(corpo);
    }

    public virtual string GetCorpo()
    {
        return corpo;
    }

    public virtual void SetCorpo(string corpo)
    {
        this.corpo = MethodBuilder.FormatCode(corpo);
    }

    public virtual string ToString()
    {
        var codigo = "";
        codigo += "while(" + condição + "){\\n\\n";
        codigo += "\\t" + corpo;
        codigo += "}\\n";
        return codigo;
    }
}